addon/metrics-adapters/google-tag-manager.js

```diff
@@ -0,0 +1,92 @@
+import Ember from 'ember';
+import canUseDOM from '../utils/can-use-dom';
+import objectTransforms from '../utils/object-transforms';
+import BaseAdapter from './base';
+
+const {
+  String: emberString,
+  assert,
+  get,
+  merge,
+  set,
+  $
+} = Ember;
+
+const {
+  capitalize
+} = emberString;
+
+const {
+  compact
+} = objectTransforms;
+
+export default BaseAdapter.extend({
+  dataLayer: 'dataLayer',
```

Comment:
that would trigger a double declaration error, i could swap it and do a check to see if it's `undefined` but then `trackEvent` and `trackPage` will not be able to get the custom name when calling `window.dataLayer`

i still have to save it as a property